alfresco-search/src/test/java/org/alfresco/solr/tracker/CascadeTrackerTest.java

```diff
@@ -0,0 +1,162 @@
+
+        indexTransaction(txn, list(folderNode,childFolderNode, fileNode), list(folderMetaData, childFolderMetaData, fileMetaData));
+
+        builder = new BooleanQuery.Builder();
+        builder.add(new BooleanClause(new TermQuery(new Term(QueryConstants.FIELD_SOLR4_ID, "TRACKER!STATE!TX")),
+            BooleanClause.Occur.MUST));
+        builder.add(new BooleanClause(LegacyNumericRangeQuery
+            .newLongRange(QueryConstants.FIELD_S_TXID, txn.getId(), txn.getId() + 1, true, false),
+            BooleanClause.Occur.MUST));
+        waitForQuery = builder.build();
+        waitForDocCount(waitForQuery, 1, MAX_WAIT_TIME);
+
+        /* Update the folder */
+        Transaction txn1 = getTransaction(0, 1);
+        folderMetaData.getProperties()
+            .put(ContentModel.PROP_CASCADE_TX, new StringPropertyValue(Long.toString(txn1.getId())));
+        folderMetaData.getProperties().put(ContentModel.PROP_NAME, new StringPropertyValue("folder2"));
+        folderNode.setTxnId(txn1.getId());
+        folderMetaData.setTxnId(txn1.getId());
+
+        //Change the ancestor on the file just to see if it's been updated
+        NodeRef nodeRef = new NodeRef(new StoreRef("workspace", "SpacesStore"), createGUID());
+        childFolderMetaData.setAncestors(ancestors(nodeRef));
+        fileMetaData.setAncestors(ancestors(nodeRef));
+
+        indexTransaction(txn1, list(folderNode), list(folderMetaData));
+
+        //Check for the TXN state stamp.
+        builder = new BooleanQuery.Builder();
+        builder.add(new BooleanClause(new TermQuery(new Term(QueryConstants.FIELD_SOLR4_ID, "TRACKER!STATE!TX")),
+            BooleanClause.Occur.MUST));
+        builder.add(new BooleanClause(LegacyNumericRangeQuery
+            .newLongRange(QueryConstants.FIELD_S_TXID, txn1.getId(), txn1.getId() + 1, true, false),
+            BooleanClause.Occur.MUST));
+        waitForDocCount(builder.build(), 1, MAX_WAIT_TIME);
+
+        TermQuery termQuery1 = new TermQuery(new Term(QueryConstants.FIELD_ANCESTOR, nodeRef.toString()));
+
+        waitForDocCount(termQuery1, 2, MAX_WAIT_TIME);
+
+        //child folder and grandchild document must be updated
+        ModifiableSolrParams params = new ModifiableSolrParams();
+        params.add("q", QueryConstants.FIELD_ANCESTOR + ":\"" + nodeRef.toString() + "\"");
+        params.add("qt", "/afts");
+        params.add("start", "0");
+        params.add("rows", "6");
+        params.add("sort", "id asc");
+        params.add("fq", "{!afts}AUTHORITY_FILTER_FROM_JSON");
+        SolrServletRequest req = areq(params,
+            "{\"locales\":[\"en\"], \"templates\": [{\"name\":\"t1\", \"template\":\"%cm:content\"}], \"authorities\": [ \"mike\"], \"tenants\": [ \"\" ]}");
+        assertQ(req, "*[count(//doc)=2]", "//result/doc[1]/long[@name='DBID'][.='" + childFolderNode.getId() + "']", "//result/doc[2]/long[@name='DBID'][.='" + fileNode.getId() + "']");
+    }
+
+    private Acl initAcls() throws Exception
```

Comment:
Please could we have some Javadoc for this method.  What ACLs specifically does it create?